File: CS/MyRadioGroup/MyRadioGroupItem.cs
using System;
using System.ComponentModel;
using DevExpress.XtraEditors.Controls;




namespace DevExpress.MyControl
{
   [TypeConverter(typeof(MyRadioGroupItemTypeConverter))]
    public class MyRadioGroupItem : RadioGroupItem
    {
        private string toolTip;


        [Browsable(true), Description("Gets or sets tooltip of the item.")]
        public string ToolTip
        {
            get { return toolTip; }
            set
            {
            	toolTip = value;
                this.ItemChanged();
            }
        }



		public MyRadioGroupItem() 
           : this(null, string.Empty, true, string.Empty)  
        {
        }



        public MyRadioGroupItem(object value, [Localizable(true)]string description) 
            : this(value, description, true, string.Empty) 
        { 
        }



        public MyRadioGroupItem(object value, [Localizable(true)]string description, bool enabled)
            : this(value, description, enabled, string.Empty)
        {
        }



        public MyRadioGroupItem(object value, [Localizable(true)]string description, bool enabled, string toolTip)
            : base(value, description, enabled)
        {
            this.toolTip = toolTip;
        }



        public override object Clone() 
        {
            return new MyRadioGroupItem(Value, Description, Enabled, ToolTip); 
        }
    }
}

File: VB/TestMyRadioGroup/MyRadioGroupItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Controls




Namespace TestMyRadioGroup
	<TypeConverter(GetType(MyRadioGroupItemTypeConverter))>
	Public Class MyRadioGroupItem
		Inherits RadioGroupItem
		Private toolTip_Renamed As String


		<Browsable(True), Description("Gets or sets tooltip of the item.")>
		Public Property ToolTip() As String
			Get
				Return toolTip_Renamed
			End Get
			Set(ByVal value As String)
				toolTip_Renamed = value
				Me.ItemChanged()
			End Set
		End Property



		Public Sub New()
			Me.New(Nothing, String.Empty, True, String.Empty)
		End Sub



		Public Sub New(ByVal value As Object, <Localizable(True)> ByVal description As String)
			Me.New(value, description, True, String.Empty)
		End Sub



		Public Sub New(ByVal value As Object, <Localizable(True)> ByVal description As String, ByVal enabled As Boolean)
			Me.New(value, description, enabled, String.Empty)
		End Sub



		Public Sub New(ByVal value As Object, <Localizable(True)> ByVal description As String, ByVal enabled As Boolean, ByVal toolTip As String)
			MyBase.New(value, description, enabled)
			Me.toolTip_Renamed = toolTip
		End Sub



		Public Overrides Function Clone() As Object
			Return New MyRadioGroupItem(Value, Description, Enabled, ToolTip)
		End Function
	End Class
End Namespace
